src/main/java/org/radarcns/management/web/rest/SubjectResource.java

```diff
@@ -289,6 +296,53 @@
         return ResponseUtil.wrapOrNotFound(Optional.ofNullable(subjectDto));
     }
 
+    /**
+     * GET  /subjects/:login/revisions : get all revisions for the "login" subject.
+     *
+     * @param login the login of the subjectDTO for which to retrieve the revisions
+     * @return the ResponseEntity with status 200 (OK) and with body the subjectDTO, or with status
+     *         404 (Not Found)
+     */
+    @GetMapping("/subjects/{login:" + Constants.ENTITY_ID_REGEX + "}/revisions")
+    @Timed
+    public ResponseEntity<List<RevisionDTO>> getSubjectRevisions(
+            @ApiParam Pageable pageable, @PathVariable String login) throws NotAuthorizedException {
+        log.debug("REST request to get revisions for Subject : {}", login);
+        SubjectDTO subject = subjectService.getLatestRevision(login);
+        Page<RevisionDTO> page = revisionService.getRevisionsForEntity(pageable,
+                subjectMapper.subjectDTOToSubject(subject));
+        SubjectDTO blank = new SubjectDTO();
+        RadarToken token = getJWT(servletRequest);
+        // iterate the revisions, if we don't have read permission for the subject in a given
+        // revision, change the subject with a 'blank' subject so no information gets leaked
```

Comment:
I would rather have another API to request revisions of subjects based on project, having one API expose partial information depends on the permission. For wrong project request we should give 401 instead of blank subject to maintain the consistency.

Reply:
The problem with such an API is that a client would have to iterate over all known projects to get a full history for a subject. 

I was thinking of the scenario where a subject is moved to another project, perhaps even multiple times. If i'm a project admin for 2 out of the 4 projects a subject had been a member of, i should be able to see the changes to the subject, only when the subject was registered to those two projects i have access to. For changes that occurred when the subject was registered to the other two projects, the rationale is to show there has been a change, so still show the revision, but not show the contents, that is where the 'blank' subject comes in. Otherwise the history this method returns is a false one.

Of course this discussion becomes pointless if a project admin can not access any historical data, since sys admins have access to all information anyway. We'll ask the wider team what roles should be able to see historical data.